packages/jest-matchers/src/index.js

```diff
@@ -10,27 +10,37 @@
 
 'use strict';
 
-import type {ThrowingMatcherFn, ExpectationResult} from '../types';
+import type {
+  Expect,
+  ExpectationResult,
+  ExpectationObject,
+  MatchersObject,
+  RawMatcherFn,
+  ThrowingMatcherFn,
+} from '../types';
 
 const matchers = require('./matchers');
+const GLOBAL_MATCHERS_OBJECT_KEY = '_JEST_MATCHERS';
+global[GLOBAL_MATCHERS_OBJECT_KEY] = {};
```

Comment:
my thoughts were to use `jest.addMatchers({toBeSomething: () => {...}});` yeah.

then in `jest-jasmine-adapter` we define a jasmine.addMatchers = (obj) => {jest.addAddmatchers(obj)});` function to make it compatible with jasmine too. (or making jasmine compatible with jest)

so no direct accessing of the global property.

I really like the idea of using a symbol though